vroom/shfmt.vroom

```diff
@@ -0,0 +1,85 @@
+The built-in shfmt formatter knows how to format shell code (POSIX/Bash/mksh).
+If you aren't familiar with basic codefmt usage yet, see main.vroom first.
+
+We'll set up codefmt and configure the vroom environment, then jump into some
+examples.
+
+  :source $VROOMDIR/setupvroom.vim
+
+  :let g:repeat_calls = []
+  :function FakeRepeat(...)<CR>
+  |  call add(g:repeat_calls, a:000)<CR>
+  :endfunction
+  :call maktaba#test#Override('repeat#set', 'FakeRepeat')
+
+  :call codefmt#SetWhetherToPerformIsAvailableChecksForTesting(0)
+
+
+The shfmt formatter expects the shfmt executable to be installed on your system.
+
+  % f()
+  :FormatCode shfmt
+  ! shfmt .*
+  $ f()
+
+The name or path of the shfmt executable can be configured via the
+shfmt_executable flag if the default of "shfmt" doesn't work.
+
+  :Glaive codefmt shfmt_executable='myshfmt'
+  :FormatCode shfmt
+  ! myshfmt .*
+  $ f()
+  :Glaive codefmt shfmt_executable='shfmt'
+
+
+You can format any buffer with shfmt specifying the formatter explicitly.
```

Comment:
Could you also include a realistic case for failure handling to demonstrate the quickfix stuff is working?

Reply:
Done.